### PR TITLE
adding the reusable sidebar

### LINKED ISSUE #407: [FRONTEND] Reusable Sidebar component
🚨 **Attention: Please read the guidelines before applying; otherwise, you will be ignored.** 🚫📋
0. Please, join the telegram group to keep in touch https://t.me/+m0IGRiBvLtsyMzIy 
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Please create reusable Sidebar component (should be put into `ui/components/Sidebar` folder with own sidebar.css file)
3. We already have some example of two sidebars ([TableOfContents](https://github.com/djeck1432/spotnet/blob/main/frontend/src/components/TableOfContent/TableOfContents.jsx) and [VaultLayout](https://github.com/djeck1432/spotnet/blob/main/frontend/src/pages/vault/VaultLayout.jsx). They should be replaced with a single one, that can accept props and based of them properly function. (you can take some styles from there as a base to your implementation if you want, if not - create it from scratch)
4. Apart from replacing old components (in **Vault/Withdraw/Documentation/Overview** pages), new Sidebar component should be added to **Dashboard** page. See [Figma](https://www.figma.com/design/WZGStY327tqwLAEhaIXccY/ZkLend-Position?node-id=1672-2595&t=C42K4OFLgvfcrogg-4)
5. Sidebar should be reused both for such case as on Dahsboard page, where we have links to other pages, as well as [this one](https://www.figma.com/design/WZGStY327tqwLAEhaIXccY/ZkLend-Position?node-id=47-1439&t=C42K4OFLgvfcrogg-4), which should actually scroll to a corresponding item on a page by its id(hash), e.g. #overview. (also should be added to url)
6. Based on props Sidebar will dynamically render **title** (optional) (as you can see for Overview page, **Content** is title) or it can be w/o title (as for Dashboard). Each sidebar should accept array of `items` (required), each item should have its own attributes, such as `icon`, `name` and `link`. Depending if it is a link to another page or #hash-link, clicking on item should perform correct action. 
7. Items can have children items (example in Overview page, **How It works** item), in that case clicking on item should not do any action, rather it should expand and show its sub items, which has the same `name` and `link` attribute as a parent item. 
8. Make sure currently selected item is highlighted like in Figma.
9. Sidebar should be visible for desktop screens (>1024px) and when page is scrolled, it should stick to the top(below header), so all content is visible (you can see how it works in current implementation and preserve this logic)
10. After creating this component, please use it on: /dashboard, /documentation, /overview, /withdraw, /stake and make sure it works correctly for each state (link to another page, scroll to item, expand sub-items).

### PR DESCRIPTION
**Pull request by Dprof-in-tech**
closes #407

### CODE REVIEW

**Comment by whateverfw** [comment]
@Dprof-in-tech seems like it is not finished yet?  or can I review already?

**Comment by Dprof-in-tech** [comment]
Please give me a couple of minutes so I can push the remaining updates .
I’ll send you a telegram dm once I do that.

File: frontend/src/pages/spotnet/dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -134,78 +138,109 @@ export default function Component({ telegramId }) {
...
-                  onClick={() => setIsCollateralActive(false)}
-                  className={`tab ${!isCollateralActive ? 'active borrow' : ''}`}
-                >
-                  <BorrowIcon className="tab-icon" />
-                  <span className="tab-title">Borrow</span>
-                </button>
-                <div className="tab-indicator-container">
-                  <div className={`tab-indicator ${isCollateralActive ? 'collateral' : 'borrow'}`} />
+    <div className="dashboard">
+      <Sidebar title={'Dashboard'} items={dashboardItems} className="sidebar-docs-sticky" />
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/Dashboard.jsx:170]: dashboard should not have title (just like in figma)

File: frontend/src/pages/spotnet/documentation/Documentation.jsx [hunk 2]
```diff
@@ -89,7 +91,8 @@ const Documentation = () => {
 
   return (
     <div className="documentation-page">
-      <TableOfContents tabelTitle={"Table Of Content"} defaultActiveId={"introduction"} headerHeight={80} items={tableOfContents} />
+      {/* <TableOfContents tabelTitle={"Table Of Content"} defaultActiveId={"introduction"} headerHeight={80} items={tableOfContents} /> */}
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/documentation/Documentation.jsx:94]: please remove commented parts

File: frontend/src/components/ui/Components/Sidebar/Sidebar.css
Comment (whateverfw) [inline frontend/src/components/ui/Components/Sidebar/Sidebar.css:1]: item's color do not match figma, should be `#83919F` in default state and `#F0F0F0` in active state

also please make sure left padding and vertical padding between items are according to figma

File: frontend/src/pages/vault/VaultLayout.jsx [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+    id: 'withdraw',
+    name: 'Withdraw',
+    link: '/withdraw',
+    icon: '•'
+  }
+  ];
   return (
     <div className="layout">
-      <aside className="sidebar">
+      {/* <aside className="sidebar">
```
Comment (whateverfw) [inline frontend/src/pages/vault/VaultLayout.jsx:29]: remove commented code

File: frontend/src/pages/vault/VaultLayout.jsx [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+ import './vaultLayout.css';
+import Sidebar from 'components/ui/Components/Sidebar/Sidebar';
 
 export function VaultLayout({ children }) {
+  const vaultItems = [
+    {
+      id: 'home',
+      name: 'Home',
+      link: '/',
+      icon: '•'
```
Comment (whateverfw) [inline frontend/src/pages/vault/VaultLayout.jsx:12]: image

Also, title styles do not match Figma

File: frontend/src/pages/spotnet/documentation/Documentation.jsx [hunk 5]
```diff
@@ -12,9 +13,10 @@ const Documentation = () => {
 
 
   const tableOfContents = [
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/documentation/Documentation.jsx:15]: image

- Icorrect title styles
- no icons near the items and seems like sub-items do not work correctly
- when clicking `Overview` I can't select any other item anymore
- when I click getting started, all items below is highlighted, also those items can't be directly selected

**Review by whateverfw — CHANGES_REQUESTED** [review]
@Dprof-in-tech Please resolve comments
Also sync with main. Notice that there were changes to folder structure and file's locations. Please merge it to your branch and make sure your changes is not overriden and applied correctly

**Comment by Dprof-in-tech** [comment]
I’m on it chief

**Comment by djeck1432** [comment]
@Dprof-in-tech Please, resolve conflict

**Comment by Dprof-in-tech** [comment]
fixed ser

**Comment by djeck1432** [comment]
@Dprof-in-tech Please, fix test cases

File: frontend/src/components/ui/Components/Sidebar/Sidebar.jsx
Comment (whateverfw) [inline frontend/src/components/ui/Components/Sidebar/Sidebar.jsx:1]: it should be located in `components/ui/sidebar` folder, as well as css file

**Review by whateverfw — CHANGES_REQUESTED** [review]
please fix ci issues and resolve earlier mentioned comments

**Review by djeck1432 — APPROVED** [review]
Partially reward, as far task is not finished

**Pull request merged**